### PR TITLE
Merge RUN_E2E_TESTS and RUN_KITCHEN_TESTS variables

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Merge the two previous CI variables we had `RUN_E2E_TESTS` and `RUN_KITCHEN_TESTS` into a single variable `RUN_TESTS` that has 3 states
- "off": No tests should be run
- "auto": Tests are run automatically according to other conditions on jobs. For example on installer tests, it will make the tests run a subset of the OS we test on PR. For e2e tests, they will be run only if they are impacted by some changes, otherwise they are manual
- "on": Force to run all the tests

This is a first PR to simplify the CI, another PR will introduce a second variable to play on test execution. `RUN_UNIT_TESTS` that will impact the execution of unit tests. `on`, `off` and `smart`

The following PR update Doggo to have these changes: <inserted once PR is created>

### Motivation

Simplify the CI

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -204,7 +204,7 @@ variables:
   CLANG_LLVM_VER: 12.0.1
   KERNEL_MATRIX_TESTING_X86_AMI_ID: "ami-0c54d42f8f4180b0c"
   KERNEL_MATRIX_TESTING_ARM_AMI_ID: "ami-021f04c00ecfa8590"
-  RUN_E2E_TESTS: "manual" # Should be "auto", "manual", "off" it will change the trigger condition for new-e2e tests on branch != main
+  RUN_E2E_ESTS: "auto" # Should be "off", "auto" or "on" it will change the trigger condition for new-e2e tests on branch != main
```
Comment (chouetz) [inline .gitlab-ci.yml:207]:
```suggestion
  RUN_E2E_TESTS: "auto" # Should be "off", "auto" or "on" it will change the trigger condition for new-e2e tests on branch != main
```

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -277,31 +277,26 @@ variables:
 .if_not_run_all_builds: &if_not_run_all_builds
   if: $CI_COMMIT_BRANCH != "main" && $DEPLOY_AGENT != "true" && $RUN_ALL_BUILDS != "true"
 
-# Rule to trigger test kitchen setup, run, and cleanup.
+# Rule to trigger test setup, run, and cleanup.
```
Comment (chouetz) [inline .gitlab-ci.yml:280]:
```suggestion
# Rule to trigger installer tests
```
It seems this comment block is only about the new `if_installer_tests` rule, as such this line is probably useless

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -277,31 +277,26 @@ variables:
...
-  if: ($CI_COMMIT_BRANCH == "main"  || $DEPLOY_AGENT == "true" || $RUN_KITCHEN_TESTS == "true") && $RUN_KITCHEN_TESTS != "false"
-
-# Rules to trigger default kitchen tests.
-# Some of the kitchen tests are run on all pipelines by default. They can only be disabled
-# by setting RUN_KITCHEN_TESTS to false.
-.if_default_kitchen: &if_default_kitchen
-  if: $RUN_KITCHEN_TESTS != "false"
+# - on main and deploy pipelines, installer tests are run
+# - on branch pipelines, installer tests are run on a subset of the OSes we test
+# RUN_E2E_TESTS can be set to on to force all the installer tests to be run on a branch pipeline.
```
Comment (chouetz) [inline .gitlab-ci.yml:284]: As `RUN_E2E_TESTS` enforces `installer_tests`, should we name it `RUN_ALL_E2E_TESTS`? I'm not sure if there are dedicated variables for installer/not installer e2e tests

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -277,31 +277,26 @@ variables:
...
 .if_testing_cleanup: &if_testing_cleanup
   if: $TESTING_CLEANUP == "true"
 
-.if_not_e2e: &if_not_e2e
-  if: $RUN_KITCHEN_TESTS == "false"
+.if_run_e2e_tests: &if_run_e2e_tests
+  if: $RUN_E2E_TESTS == "on"
 
-.if_auto_e2e: &if_auto_e2e
+# When RUN_E2E_TESTS is set to "auto", we run only a subset of installer tests on branch pipelines. For e2e tests it will run tests only if they are impacted by some modifications made on the PR.
```
Comment (chouetz) [inline .gitlab-ci.yml:295]:
```suggestion
# When RUN_E2E_TESTS is set to "auto", we run: 
# - a subset of installer tests on branch pipelines. 
# - agent-related e2e tests based on files changed in the PR.
```

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -792,7 +787,7 @@ workflow:
   - when: on_success
 
 .except_no_tests_no_deploy:
-  - if: $RUN_KITCHEN_TESTS == "false" && $DEPLOY_AGENT == "false" && $RUN_E2E_TESTS == "off"
+  - if:  $DEPLOY_AGENT == "false" && $RUN_TESTS == "off"
```
Comment (chouetz) [inline .gitlab-ci.yml:790]:
```suggestion
  - if: $DEPLOY_AGENT == "false" && $RUN_E2E_TESTS == "off"
```

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -859,45 +854,39 @@ workflow:
 .on_kitchen_tests_a6:
```
Comment (chouetz) [inline .gitlab-ci.yml:854]:
```suggestion
.on_installer_e2e_tests_a6:
```
But this could deserve a dedicated PR I guess, as it requires a modification of this reference as well

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -859,45 +854,39 @@ workflow:
...
   - <<: *if_not_version_6
     when: never
-  - <<: *if_kitchen
+  - <<: *if_installer_tests
 
 .on_kitchen_tests_a6_always:
```
Comment (chouetz) [inline .gitlab-ci.yml:859]:
```suggestion
.on_installer_e2e_tests_a6_always:
```
argh, sorry, I cannot stop myself from doing this suggestion :D

File: .gitlab-ci.yml [hunk 8]
```diff
@@ -859,45 +854,39 @@ workflow:
...
   - <<: *if_not_version_6
     when: never
-  - <<: *if_kitchen
+  - <<: *if_installer_tests
     when: always
 
 .on_all_kitchen_builds_a6:
```
Comment (chouetz) [inline .gitlab-ci.yml:865]:
```suggestion
.on_installer_e2e_tests_all_builds_a6:
```

File: .gitlab-ci.yml [hunk 9]
```diff
@@ -859,45 +854,39 @@ workflow:
...
-  - <<: *if_kitchen
-
-.on_kitchen_tests_a7_always:
-  - <<: *if_not_version_7
-    when: never
-  - <<: *if_kitchen
-    when: always
+  - <<: *if_installer_tests
 
 .on_all_kitchen_builds_a7:
```
Comment (chouetz) [inline .gitlab-ci.yml:877]:
```suggestion
.on_installer_e2e_test_all_builds_a7:
```

**Review by chouetz — CHANGES_REQUESTED** [review]
You can probably ignore the rename of the "public" rules. However I think there are some typos to be fixed

File: .gitlab-ci.yml (hunk 3 above)
Comment (KevinFairise2) [reply .gitlab-ci.yml:284]: We had dedicated variables for E2E and installer tests `RUN_E2E_TESTS` and `RUN_KITCHEN_TESTS` the goal of this PR is to merge the two. I think we no longer want to make a difference between what we call E2E tests and the installer tests that are also E2E tests

File: .gitlab-ci.yml (hunk 6 above)
Comment (KevinFairise2) [reply .gitlab-ci.yml:856]: Yes I would prefer to have a dedicated PR for that. This one change the logic and we could have another that change only the naming

File: .gitlab-ci.yml [hunk 10]
```diff
@@ -859,45 +854,39 @@ workflow:
...
   - <<: *if_not_version_7
     when: never
   - <<: *if_not_run_all_builds
     when: never
-  - <<: *if_kitchen
+  - <<: *if_installer_tests
 
 .on_all_new-e2e_tests_a7:
```
[the remark this answers is not in the document]
Comment (KevinFairise2) [reply .gitlab-ci.yml:886]: Yes it is, it was introduced when I started to migrate from `kitchen` to `new-e2e` the installer tests. This should probably be merged as soon as we rename the rule

File: .gitlab-ci.yml (hunk 7 above)
Comment (KevinFairise2) [reply .gitlab-ci.yml:861]: I promise I will do a follow up PR to fix this :D 
https://datadoghq.atlassian.net/jira/software/c/projects/APL/boards/5339

File: .gitlab-ci.yml (hunk 3 above)
Comment (KevinFairise2) [reply .gitlab-ci.yml:284]: I think a good compromise is to keep the variable `RUN_E2E_TESTS` but rename the alias. `if_run_e2e_tests` into `if_run_all_e2e_tests`. Because the consequence of having `RUN_E2E_TESTS` to `on` is that we run all the e2e tests

**Review by chouetz — APPROVED** [review]
